File: util/talk/internal/server/internal_evidence_test.go
package server

import (
	"strings"
	"testing"
)

func TestInternalEvidenceOnly(t *testing.T) {
	marker := `[Historical tool evidence: web_fetch; archive_id=524; use context_read for original] {"content":"private context"}`
	input := "답변\n" + marker + "\n다음 설명\n" + `{"valid":"JSON"}` + "\n[일반 대괄호]"
	out := cleanInternalEvidence(input)
	if strings.Contains(out, "private context") || !strings.Contains(out, `{"valid":"JSON"}`) || !strings.Contains(out, "다음 설명") || !strings.Contains(out, "[일반 대괄호]") {
		t.Fatal(out)
	}
	code := "```text\n" + marker + "\n```"
	if cleanInternalEvidence(code) != code {
		t.Fatal("code example changed")
	}
}

File: util/talk/web/src/lib/internal-evidence.test.js
import test from 'node:test';
import assert from 'node:assert/strict';
import { stripInternalEvidence } from './internal-evidence.js';
test('hide reference records, retain answer JSON and code', () => {
 const line = '[Historical tool evidence: web_fetch; archive_id=524; use context_read for original] {"content":"internal"}';
 assert.equal(stripInternalEvidence(`Answer\n${line}\n{"normal":true}\n[normal]`),'Answer\n{"normal":true}\n[normal]');
 const code = '```text\n'+line+'\n```'; assert.equal(stripInternalEvidence(code),code);
 for(let n=12;n<line.indexOf(']');n++) assert.equal(stripInternalEvidence(line.slice(0,n)),'');
});
